### PR TITLE
Add 'useCustomMailer' and 'sendExistingUserEmail' config options

### PR DESCRIPTION
**Pull request by maxcodefaster**
Adds option for using a custom mailer by listening to the emitted events, instead of using the built in mailer.

### CODE REVIEW

File: src/user.ts [hunk 1]
```diff
@@ -956,7 +960,7 @@ export class User {
   }
 
   private async sendModifiedPasswordEmail(user: SlUserDoc, req): Promise<void> {
-    if (this.config.local.sendPasswordChangedEmail) {
+    if (this.config.local.sendPasswordChangedEmail && !this.config.mailer.useCustomMailer) {
```
Comment (ikosta) [inline src/user.ts:963]: Here is no event emited

File: src/user.ts [hunk 2]
```diff
@@ -471,7 +473,7 @@ export class User {
     );
     const result = await this.userDB.insert(finalNewUser);
     newUser._rev = result.rev;
-    if (this.config.local.sendConfirmEmail) {
+    if (this.config.local.sendConfirmEmail && !this.config.mailer.useCustomMailer) {
```
Comment (ikosta) [inline src/user.ts:476]: Here is no event emited

File: src/user.ts [hunk 3]
```diff
@@ -1088,14 +1094,16 @@ export class User {
...
       };
-      await this.mailer.sendEmail(
-        'confirmEmailChange',
-        user.unverifiedEmail.email,
-        {
-          req: req,
-          user: user
-        }
-      );
+      if (!this.config.mailer.useCustomMailer) {
```
Comment (ikosta) [inline src/user.ts:1097]: Here is no event emited

**Review by ikosta — CHANGES_REQUESTED** [review]

File: src/user.ts (hunk 1 above)
Comment (maxcodefaster) [reply src/user.ts:963]: The event is emitted here:

https://github.com/perfood/couch-auth/blob/325148330d65f4950ba46ec08b637e07eff25f4d/src/user.ts#L959

File: src/user.ts (hunk 2 above)
Comment (maxcodefaster) [reply src/user.ts:476]: The event is emitted here:

https://github.com/perfood/couch-auth/blob/325148330d65f4950ba46ec08b637e07eff25f4d/src/user.ts#L421

File: src/user.ts (hunk 3 above)
Comment (maxcodefaster) [reply src/user.ts:1097]: The event is emitted here:

https://github.com/perfood/couch-auth/blob/325148330d65f4950ba46ec08b637e07eff25f4d/src/user.ts#L1116

**Review by ikosta — APPROVED** [review]
LGTM

**Pull request merged**